testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -202,292 +203,278 @@ def iter_ts(self, i):
+        assert_timestep_almost_equal(ts, ref.first_frame,
+                                     decimal=ref.prec)
+
+    def test_rename_aux(self, ref, reader):
+        reader.rename_aux('lowf', 'lowf_renamed')
         # data should now be in aux namespace under new name
-        assert_equal(self.reader.ts.aux.lowf_renamed,
-                     self.ref.aux_lowf_data[0])
+        assert_equal(reader.ts.aux.lowf_renamed,
+                    ref.aux_lowf_data[0])
         # old name should be removed
-        assert_raises(AttributeError, getattr, self.reader.ts.aux, 'lowf')
+        assert_raises(AttributeError, getattr, reader.ts.aux, 'lowf')
         # new name should be retained
-        next(self.reader)
-        assert_equal(self.reader.ts.aux.lowf_renamed,
-                     self.ref.aux_lowf_data[1])
+        next(reader)
+        assert_equal(reader.ts.aux.lowf_renamed,
+                     ref.aux_lowf_data[1])
 
-    def test_iter_as_aux_highf(self):
+    def test_iter_as_aux_highf(self, ref, reader):
         # auxiliary has a higher frequency, so iter_as_aux should behave the
         # same as regular iteration over the trjectory
-        for i, ts in enumerate(self.reader.iter_as_aux('highf')):
-            assert_timestep_almost_equal(ts, self.ref.iter_ts(i),
-                                         decimal=self.ref.prec)
+        for i, ts in enumerate(reader.iter_as_aux('highf')):
+            assert_timestep_almost_equal(ts, ref.iter_ts(i),
+                                         decimal=ref.prec)
 
-    def test_iter_as_aux_lowf(self):
+    def test_iter_as_aux_lowf(self, ref, reader):
         # auxiliary has a lower frequency, so iter_as_aux should iterate over
         # only frames where there is a corresponding auxiliary value
-        for i, ts in enumerate(self.reader.iter_as_aux('lowf')):
+        for i, ts in enumerate(reader.iter_as_aux('lowf')):
             assert_timestep_almost_equal(ts,
-                                         self.ref.iter_ts(self.ref.aux_lowf_frames_with_steps[i]),
-                                         decimal=self.ref.prec)
+                                         ref.iter_ts(ref.aux_lowf_frames_with_steps[i]),
+                                         decimal=ref.prec)
 
 
 class BaseWriterTest(object):
-    def __init__(self, reference):
-        self.ref = reference
-        self.tmpdir = tempdir.TempDir()
-        self.reader = self.ref.reader(self.ref.trajectory)
-
-    def tmp_file(self, name):
-        return self.tmpdir.name + name + '.' + self.ref.ext
-
-    def test_write_trajectory_timestep(self):
-        outfile = self.tmp_file('write-timestep-test')
-        with self.ref.writer(outfile, self.reader.n_atoms) as W:
-            for ts in self.reader:
+    def tmp_file(self, name, ref, tmpdir):
+        return tmpdir.name + name + '.' + ref.ext
+
+    def test_write_trajectory_timestep(self,ref, reader, tmpdir):
```

Comment:
I'm confused here. Is tmpdir here using the tmpdir-fixture of pytest? Because later you use `tmpdir.name` but the path.local variable that the tmpdir-fixture returns doesn't have a `name` attribute.